From 748ff59e12e38c9e4d6ec69ed92a1d1bde5a2f3c Mon Sep 17 00:00:00 2001
From: Maciej Sobkowski <maciej@sobkow.ski>
Date: Tue, 29 Aug 2017 12:31:54 +0200
Subject: [PATCH] Add PRU DMA resources

---
 include/pru_types.h | 18 ++++++++++++++++++
 include/rsc_types.h |  1 +
 2 files changed, 19 insertions(+)

diff --git a/include/pru_types.h b/include/pru_types.h
index c8ab8b1..1fbec09 100644
--- a/include/pru_types.h
+++ b/include/pru_types.h
@@ -36,6 +36,7 @@
 
 /* Custom Resource info: Must match drivers/remoteproc/pru_rproc.h */
 #define TYPE_PRU_INTS		1
+#define TYPE_PRU_DMA		2
 
 /**
  * struct ch_map - sysevts-to-channel mapping
@@ -80,4 +81,21 @@ struct fw_rsc_custom_ints {
 	struct ch_map *event_channel;
 };
 
+#define DMA_MAX_CHANNELS 10
+
+struct dma_ch {
+       //buffer address (filled by host), EDMA channel, EDMA PaRAM slot, buffer size, completion
+       uint32_t buf_addr;
+       uint32_t buf_size;
+       uint32_t edma_channel;
+       uint32_t param_slot;
+       uint32_t notify_completion;
+} __attribute__((__packed__));
+
+struct fw_rsc_custom_dma_ch {
+       uint32_t version;
+       uint32_t dma_ch_size;
+       struct dma_ch dma_ch[DMA_MAX_CHANNELS];
+} __attribute__((__packed__));
+
 #endif /* _PRU_TYPES_H_ */
diff --git a/include/rsc_types.h b/include/rsc_types.h
index 755ab60..a6fd65b 100644
--- a/include/rsc_types.h
+++ b/include/rsc_types.h
@@ -51,6 +51,7 @@
 union fw_custom {
 	/* add custom resources here */
 	struct fw_rsc_custom_ints pru_ints;
+	struct fw_rsc_custom_dma_ch pru_dma;
 	/* maintain reserved as the last element */
 	uint32_t reserved;
 };
-- 
2.14.1

